src/config-wrapper.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const { Request, Response } = require('@adobe/helix-fetch');
+const { cleanupHeaderValue } = require('./utils');
+const redirect = require('./RedirectConfig');
+const helix = require('./HelixConfig');
+const fstab = require('./MountConfig');
+const index = require('./IndexConfig');
+const markup = require('./MarkupConfig');
+
+const loaders = {
+  redirect,
+  helix,
+  fstab,
+  index,
+  markup,
+};
+
+async function getData(request, ...names) {
+  if (/^application\/x-www-form-urlencoded/.test(request.headers.get('content-type'))) {
+    const data = new URLSearchParams(await request.text());
+    return names.reduce((prev, name) => {
+      if (data.get(name)) {
+        // eslint-disable-next-line no-param-reassign
+        prev[name] = Number.isNaN(Number(data.get(name)))
+          ? data.get(name)
+          : Number.parseInt(data.get(name), 10);
+      }
+      return prev;
+    }, {});
+  } else if (/json/.test(request.headers.get('content-type'))) {
+    const data = await request.json();
+    const ret = names.reduce((prev, name) => {
+      if (data[name]) {
+      // eslint-disable-next-line no-param-reassign
+        prev[name] = data[name];
+      }
+      return prev;
+    }, {});
+    return ret;
+  }
+  // just return the request parameters
+  const data = new URLSearchParams(new URL(request.url).search);
+  return names.reduce((prev, name) => {
+    if (data.get(name)) {
+      // eslint-disable-next-line no-param-reassign
+      prev[name] = Number.isNaN(Number(data.get(name)))
+        ? data.get(name)
+        : Number.parseInt(data.get(name), 10);
+    }
+    return prev;
+  }, {});
+}
+
+function getToken({ headers }) {
+  if (headers.get('authorization')) {
+    return headers.get('authorization');
+  }
+  if (headers.get('x-github-token')) {
+    return `token ${headers.get('x-github-token')}`;
+  }
+  return undefined;
+}
+
+function wrap(func, required, ...configs) {
+  return async (request, context) => {
+    const { owner, repo, ref } = await getData(request, 'owner', 'repo', 'ref');
+    const {
+      transactionId,
+    } = context.invocation;
+
+    const authorization = getToken(request);
+
+    const options = {
+      headers: {
+        ...(authorization ? { authorization } : authorization),
+      },
+    };
```

Comment:
I don't like this.... what's wrong with the good old if?

```suggestion
    const options = {
      headers: {
      },
    };
    if (authorization) {
      options.headers.authorization = authorization;
    }
```